lib/js/dom.nim

```diff
@@ -1471,16 +1471,20 @@ else:
   proc createTextNode*(d: Document, identifier: cstring): Node {.importcpp.}
   proc createComment*(d: Document, data: cstring): Node {.importcpp.}
 
-proc setTimeout*(action: proc(); ms: int): TimeOut {.importc, nodecl.}
+proc setTimeout*[T](action: proc (): T; ms: int): TimeOut {.importc, nodecl.}
+proc setTimeout*(action: proc (); ms: int): TimeOut {.importc, nodecl.}
```

Comment:
```suggestion
proc setTimeout*[T: proc (): T | proc ()](action: T, ms: int): TimeOut {.importc, nodecl.}
```

This might work.

Reply:
this cannot work.

instead you can use this:
```nim
type TimeoutCallback = concept x
  x()
proc setTimeout(action: TimeoutCallback, ms: int): TimeOut {.importc, nodecl.}
```
which avoids duplicating API declarations.

ditto with other similar instances

Reply:
`ms` argument should be `SomeInteger` IMHO.

Reply:
meh, no need to be generic on this; user can do the conversion to int on their end